src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      authContext: {
+        authData: {},
+        isAuthenticated: this.isAuthenticated,
+        setCurrentUser: this.setCurrentUser,
+        setCurrentUserProfile: this.setCurrentUserProfile
+      }
+    };
+  }
+
+  componentDidMount = () => {
+    // Sync auth session stored in the localStorage with Context
+    // TBH, I'm not sure if we even need context
+    this.setCurrentUser(AuthService.getSessionAuthData());
+  };
+
+  isAuthenticated = () => {
+    const {
+      userId,
+      expiresAt,
+      auth0AccessToken,
+      graphcoolToken
+    } = this.state.authContext.authData;
+    return (
+      Date.now() < expiresAt && userId && auth0AccessToken && graphcoolToken
+    );
+  };
+
+  setCurrentUser = authData => {
+    this.setState({
+      authContext: {
+        ...this.state.authContext,
+        authData
+      }
+    });
+  };
+
+  setCurrentUserProfile = userProfile => {
+    AuthService.setAuthProfileSession(userProfile); // I don't really like this side-effect being here
+    this.setState({
```

Comment:
These calls to `setState` should probably be using the function argument [updater format](https://reactjs.org/docs/react-component.html#setstate) as the state update is based on current state:
```js
this.setState(state => ({
    authContext: {
        state.authContext,
        authData: {
          state.authContext.authData,
          userProfile
        }
    }
}));
```